includes/Core/Email_Reporting/Email_Log.php

```diff
@@ -0,0 +1,383 @@
+<?php
+/**
+ * Class Google\Site_Kit\Core\Email_Reporting\Email_Log
+ *
+ * @package   Google\Site_Kit\Core\Email_Reporting
+ * @copyright 2025 Google LLC
+ * @license   https://www.apache.org/licenses/LICENSE-2.0 Apache License 2.0
+ * @link      https://sitekit.withgoogle.com
+ */
+
+namespace Google\Site_Kit\Core\Email_Reporting;
+
+use Google\Site_Kit\Core\User\Email_Reporting_Settings as Reporting_Settings;
+use Google\Site_Kit\Core\Util\Method_Proxy_Trait;
+
+/**
+ * Registers the internal Email Reporting log storage.
+ *
+ * @since n.e.x.t
+ * @access private
+ * @ignore
+ */
+final class Email_Log {
+
+	use Method_Proxy_Trait;
+
+	/**
+	 * Post type slug.
+	 */
+	const POST_TYPE = 'gsk_email_log';
+
+	/**
+	 * Report frequency meta key.
+	 */
+	const META_REPORT_FREQUENCY = 'report_frequency';
+
+	/**
+	 * Batch ID meta key.
+	 */
+	const META_BATCH_ID = 'batch_id';
+
+	/**
+	 * Send attempts meta key.
+	 */
+	const META_SEND_ATTEMPTS = 'send_attempts';
+
+	/**
+	 * Error details meta key.
+	 */
+	const META_ERROR_DETAILS = 'error_details';
+
+	/**
+	 * Report reference dates meta key.
+	 */
+	const META_REPORT_REFERENCE_DATES = 'report_reference_dates';
+
+	/**
+	 * Registers functionality through WordPress hooks.
+	 *
+	 * @since n.e.x.t
+	 */
+	public function register() {
+		add_action( 'init', $this->get_method_proxy_once( 'register_email_log' ) );
+	}
+
+	/**
+	 * Registers the email log post type, statuses, and meta.
+	 *
+	 * @since n.e.x.t
+	 */
+	protected function register_email_log() {
+		$this->register_post_type();
+		$this->register_post_statuses();
+		$this->register_post_meta();
+	}
+
+	/**
+	 * Registers the internal email log post type.
+	 *
+	 * @since n.e.x.t
+	 */
+	protected function register_post_type() {
+		if ( post_type_exists( self::POST_TYPE ) ) {
+			return;
+		}
+
+		register_post_type(
+			self::POST_TYPE,
+			array(
+				'labels'              => array(
+					'name'               => _x( 'Email Reports', 'Post type general name', 'google-site-kit' ),
+					'singular_name'      => _x( 'Email Report', 'Post type singular name', 'google-site-kit' ),
+					'add_new'            => _x( 'Add New', 'email report', 'google-site-kit' ),
+					'add_new_item'       => __( 'Add New Email Report', 'google-site-kit' ),
+					'edit_item'          => __( 'Edit Email Report', 'google-site-kit' ),
+					'new_item'           => __( 'New Email Report', 'google-site-kit' ),
+					'view_item'          => __( 'View Email Report', 'google-site-kit' ),
+					'search_items'       => __( 'Search Email Reports', 'google-site-kit' ),
+					'not_found'          => __( 'No Email Reports found.', 'google-site-kit' ),
+					'not_found_in_trash' => __( 'No Email Reports found in Trash.', 'google-site-kit' ),
+				),
+				'public'              => false,
+				'show_ui'             => false,
```

Comment:
The default value for this parameter is the value of the `public` parameter, which is also `false`. So no need to add it explicitly. 
```suggestion
```